monorepo/mytsup/tsup.mjs

```diff
@@ -40,14 +40,15 @@ export default async (options, ourOptions) => {
       ),
       TARGET: "node",
     },
-    clean: true,
+    clean: false, // we do this ourselves so its granular
```

Comment:
This was deleting the entire output directory, which meant that typecheck ALWAYS had to run after transpile. If you ran typecheck. then did a transpile run, you would lose your typecheck output.